ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java

```diff
@@ -136,6 +137,7 @@ public Program(byte[] codeHash, byte[] ops, ProgramInvoke programInvoke, Transac
         traceListener = new ProgramTraceListener(config.vmTrace());
         this.memory = setupProgramListener(new Memory());
         this.stack = setupProgramListener(new Stack());
+        this.originalRepo = programInvoke.getRepository().getSnapshotTo(programInvoke.getRepository().getRoot());
```

Comment:
Ooops, we could have not only `RepositoryImpl` here at least in tests.
Any ideas how can we get copy of Repo here?